File: codes/kinematics/forward-kinematics/py/rotation.py
import math

def Rx(t):
    c, s = math.cos(t), math.sin(t)
    return [[1, 0, 0],
            [0, c,-s],
            [0, s, c]]

def Ry(t):
    c, s = math.cos(t), math.sin(t)
    return [[ c, 0, s],
            [ 0, 1, 0],
            [-s, 0, c]]

def Rz(t):
    c, s = math.cos(t), math.sin(t)
    return [[c,-s, 0],
            [s, c, 0],
            [0, 0, 1]]

def pretty_print(M, width=9, prec=4, label="R"):
    fmt = f"{{:{width}.{prec}f}}"
    print(f"{label} =")
    for row in M:
        print("[ " + "  ".join(fmt.format(x) for x in row) + " ]")

if __name__ == "__main__":
    axis = input("Axis (x/y/z): ").strip().lower()
    try:
        ang_deg = float(input("Angle (degrees): ").strip())
    except ValueError:
        print("Invalid angle."); raise SystemExit(1)

    t = math.radians(ang_deg)

    if axis == "x":
        R = Rx(t); lbl = f"R_x({ang_deg}°)"
    elif axis == "y":
        R = Ry(t); lbl = f"R_y({ang_deg}°)"
    elif axis == "z":
        R = Rz(t); lbl = f"R_z({ang_deg}°)"
    else:
        print("Axis must be x, y, or z."); raise SystemExit(1)

    pretty_print(R, label=lbl)

File: codes/kinematics/forward-kinematics/py/rotation.cpp
#include <cmath>
#include <array>
#include <iostream>
#include <iomanip>
#include <cctype>

using Mat3 = std::array<std::array<double,3>,3>;

Mat3 Rx(double t){
    double c = std::cos(t), s = std::sin(t);
    return {{{1, 0, 0},
             {0, c,-s},
             {0, s, c}}};
}
Mat3 Ry(double t){
    double c = std::cos(t), s = std::sin(t);
    return {{{ c, 0, s},
             { 0, 1, 0},
             {-s, 0, c}}};
}
Mat3 Rz(double t){
    double c = std::cos(t), s = std::sin(t);
    return {{{c,-s, 0},
             {s, c, 0},
             {0, 0, 1}}};
}

void pretty_print(const Mat3& R, const std::string& label, int width=10, int prec=4){
    std::cout << label << " =\n";
    std::cout << std::fixed << std::setprecision(prec);
    for(const auto& row : R){
        std::cout << "[ ";
        for(size_t j=0;j<3;++j){
            std::cout << std::setw(width) << row[j] << (j<2 ? "  " : "");
        }
        std::cout << " ]\n";
    }
}

int main(){
    char axis;
    std::cout << "Axis (x/y/z): ";
    if(!(std::cin >> axis)) return 1;
    axis = std::tolower(axis);

    double ang_deg;
    std::cout << "Angle (degrees): ";
    if(!(std::cin >> ang_deg)) return 1;

    double t = ang_deg * M_PI / 180.0;

    Mat3 R;
    std::string label;
    if(axis == 'x'){ R = Rx(t); label = "R_x(" + std::to_string(ang_deg) + "°)"; }
    else if(axis == 'y'){ R = Ry(t); label = "R_y(" + std::to_string(ang_deg) + "°)"; }
    else if(axis == 'z'){ R = Rz(t); label = "R_z(" + std::to_string(ang_deg) + "°)"; }
    else { std::cerr << "Axis must be x, y, or z.\n"; return 1; }

    pretty_print(R, label);
    return 0;
}
